airflow/contrib/operators/jenkins_job_trigger_operator.py

```diff
@@ -0,0 +1,103 @@
+    :type parameters: int
+    """
+    template_fields = ('parameters',)
+    template_ext = ('.json',)
+    ui_color = '#f9ec86'
+
+    @apply_defaults
+    def __init__(self,
+                 jenkins_connection_id="",
+                 job_name="",
+                 parameters="",
+                 sleep_time=10,
+                 *args,
+                 **kwargs):
+        super(JenkinsJobTriggerOperator, self).__init__(*args, **kwargs)
+        self.job_name = job_name
+        self.parameters = parameters
+        if sleep_time < 1:
+            sleep_time = 1
+        self.sleep_time = sleep_time
+        self.jenkins_connection_id = jenkins_connection_id
+
+    def get_hook(self):
+        return JenkinsHook(self.jenkins_connection_id)
+
+    def execute(self, context):
+        if not self.jenkins_connection_id:
+            logging.error("Please specify the jenkins connection id to use. You must create a Jenkins connection before being able to use this operator")
+            raise AirflowException("The jenkins_connection_id parameter is missing, impossible to trigger the job")
+
+        if not self.job_name:
+            logging.error("Please specify the job name to use in the job_name parameter")
+            raise AirflowException("The job_name parameter is missing, impossible to trigger the job")
+
+        logging.info('Triggering the job %s on the jenkins : %s with the parameters : %s', self.job_name, self.jenkins_connection_id, self.parameters)
+        jenkins_server = self.get_hook().get_jenkins_server()
+        build_number = jenkins_server.get_job_info(job_name=self.job_name)['nextBuildNumber']
+
+        if self.parameters:
+            if isinstance(self.parameters, basestring):
+                import ast
+                self.parameters = ast.literal_eval(self.parameters)
+            jenkins_server.build_job(job_name=self.job_name, parameters=self.parameters)
+        else:
+            jenkins_server.build_job(job_name=self.job_name)
+
+        time.sleep(self.sleep_time)
+        not_found_error_count = 0
+        keep_polling_job = True
+        while (keep_polling_job):
+            try:
+                build_info = jenkins_server.get_build_info(job_name=self.job_name, build_number=build_number)
+                if (build_info['result'] is not None):
+                    keep_polling_job = False
+                    # Check if job had errors.
+                    if build_info['result'] != 'SUCCESS':
+                        raise AirflowException('Jenkins job failed, final state : '+build_info['result']+'. Find more information on job url : '+build_info['url'])
+                else:
+                    logging.info('Waiting for job to complete : %s , build %s', self.job_name, build_number)
+                    time.sleep(self.sleep_time)
+            except jenkins.NotFoundException as err:
+                not_found_error_count = not_found_error_count + 1
+                if not_found_error_count < 10:
```

Comment:
nit: it may be better to pass in an arg like max_retry_attempt or max_wait_time instead of hard-coding retry attempts = 10, so that the total retry time is a bit more flexible and doesn't have to depend on sleep_time.

Reply:
Thanks, that's fixed :)